back/app/controllers/web_api/v1/projects_controller.rb

```diff
@@ -76,6 +76,46 @@ def create
     end
   end
 
+  def copy
+    source_project = Project.find(params[:id])
+
+    new_title_multiloc = add_suffix_to_title(source_project.title_multiloc)
+    temporary_slug = SecureRandom.urlsafe_base64(32).downcase # Use to find created project
+
+    options = {
+      local_copy: true,
+      include_ideas: false,
+      anonymize_users: false,
+      new_slug: temporary_slug,
+      new_title_multiloc: new_title_multiloc,
+      timeline_start_at: Time.now.to_s,
+      new_publication_status: 'draft'
+    }
+
+    template = AdminApi::ProjectCopyService.new.export source_project, **options
+    folder_id = ProjectFolders::Folder.find(source_project.folder_id) if source_project.folder_id
+    AdminApi::ProjectCopyService.new.import(template, folder: folder_id)
+
+    # We don't search for an exact match as ProjectCopyService#import may well add a numbered suffix.
+    @project = Project.where('slug LIKE ?', "%#{temporary_slug}%").first
```

Comment:
`AdminApi::ProjectCopyService.new.import` should already return the newly created project, you don't need to query it again.

Reply:
How? It's importing a template of models?

Reply:
Also, the code
```ruby
Project.where.not(id: project_ids_before)...
project.update
```
in `ProjectCopyService` does not seem reliable and could cause bugs. If you want, you can fix it.

Reply:
> How? It's importing a template of models.

You can use this code for testing
```ruby
test = AdminApi::ProjectCopyService.new.import(template, folder: folder_id)
p test
```

Reply:
Great! I thought I tried that, but obviously I didn't!

Reply:
> in ProjectCopyService does not seem reliable and could cause bugs. If you want, you can fix it.

How is it unreliable? What might the fix be?

Reply:
> > in ProjectCopyService does not seem reliable and could cause bugs. If you want, you can fix it.
> 
> How is it unreliable? What might the fix be?
> 
> Do you mean I should use something like my UUID in slug idea to make this more robust?

Reply:
Also should I move the code in `side_fx.after_copy` into the new `LocalProjectCopyService`, or leave it where it is?

Reply:
> Also, should I move the code in side_fx.after_copy into the new LocalProjectCopyService, or leave it where it is?

I believe we usually call side_fx in controllers. So, it's fine here.

> How is it unreliable? (because another project could be created before template applied, I guess) 

Yes, indeed. It takes all the projects that were created after this line
```
      project_ids_before = Project.ids
```
What if somebody created a project between that line and `Project.where.not(id: project_ids_before)`.

> What might the fix be?

`resolve_and_apply_template` could return the project that was imported. And then we work with this project only.

Reply:
> resolve_and_apply_template could return the project that was imported. And then we work with this project only.

I think `resolve_and_apply_template` is used when applying _any_ template, including of whole tenants, so we would have to pass a flag to that method to indicate we want a return _when we are copying a project_.